src/NHibernate/Linq/Functions/EqualsGenerator.cs

```diff
@@ -42,6 +42,9 @@ public EqualsGenerator()
 					ReflectHelper.GetMethodDefinition<TimeSpan>(x => x.Equals(x)),
 					ReflectHelper.GetMethodDefinition<bool>(x => x.Equals(default(bool))),
 
+					ReflectHelper.GetMethodDefinition<object>(x => x.Equals(default(object))),
```

Comment:
This is already handled by Linq itself.

Reply:
I've double-checked and it turns it does not work.

Reply:
Remove it, and `QueryWithDefaultEqualsShouldNotThrow` fails [here](https://github.com/nhibernate/nhibernate-core/blob/5.1.2/src/NHibernate/Linq/Visitors/HqlGeneratorExpressionVisitor.cs#L514). So it does appears to be translated to another expression.